packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -215,9 +205,18 @@ class DropDownButton<T> extends StatelessComponent {
   void _showDropDown(BuildContext context, int selectedIndex, GlobalKey indexedStackKey) {
     final RenderBox renderBox = indexedStackKey.currentContext.findRenderObject();
     final Rect rect = renderBox.localToGlobal(Point.origin) & renderBox.size;
+    final NavigatorState navigator = Navigator.of(context);
```

Comment:
Can we do this working inside the `ModalPosition position` getter?  The Route should have a refernece to the navigator already, which will remove the "of" call.